Add Check Support for Java 17 Sealed Classes: IllegalType

child of #14969 
I have read check documentation: https://checkstyle.org/checks/coding/illegaltype.html#IllegalType
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> cat src/Test.java                                                 

class IllegalType { }
public class Test extends IllegalType {            // violation
    public sealed class A permits IllegalType2 { }  // expected violation
    final class IllegalType2 extends A { }
}
PS D:\CS\test> cat config.xml                                                    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
         <module name="IllegalType">
            <property name="illegalClassNames" value="IllegalType, IllegalType2"/>
        </module>

    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java 
Starting audit...
[ERROR] D:\CS\test\src\Test.java:3:27: Usage of type 'IllegalType' is not allowed. [IllegalType]
Audit done.
Checkstyle ends with 1 errors.
PS D:\CS\test> 
```


---

**Describe what you expect in detail.**

violation on the Illegal classes in the permitted list 

---

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheckTest.java
@@ -486,4 +486,17 @@ public class IllegalTypeCheckTest extends AbstractModuleTestSupport {
                 getNonCompilablePath("InputIllegalTypeWithRecordPattern.java"),
                 expected);
     }
+
+    @Test
+    public void testIllegalTypeInPermitsList() throws Exception {
+        final String[] expected = {
+            "22:52: " + getCheckMessage(MSG_KEY, "D"),
+            "27:28: " + getCheckMessage(MSG_KEY, "D"),
+            "32:27: " + getCheckMessage(MSG_KEY, "C"),
+            "35:27: " + getCheckMessage(MSG_KEY, "D"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputIllegalTypeInPermitsList.java"),
+                expected);
+    }
 }

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltype/InputIllegalTypeInPermitsList.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltype/InputIllegalTypeInPermitsList.java
@@ -1,0 +1,36 @@
+/*
+IllegalType
+validateAbstractClassNames = (default)false
+illegalClassNames = A, B, C, D
+legalAbstractClassNames = (default)
+ignoredMethodNames = (default)getEnvironment, getInitialContext
+illegalAbstractClassNameFormat = (default)^(.*[.])?Abstract.*$
+memberModifiers = (default)
+tokens = (default)ANNOTATION_FIELD_DEF, CLASS_DEF, INTERFACE_DEF, METHOD_CALL, METHOD_DEF, \
+         METHOD_REF, PARAMETER_DEF, VARIABLE_DEF, PATTERN_VARIABLE_DEF, RECORD_DEF, \
+         RECORD_COMPONENT_DEF, RECORD_PATTERN_DEF
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltype;
+
+class D { }
+
+// violation below, 'Usage of type 'D' is not allowed.'
+public class InputIllegalTypeInPermitsList extends D {
+
+    sealed class D permits B, E { } // ok, We don't ban the instantiation of illegal types
+
+    // violation below, 'Usage of type 'D' is not allowed.'
+    sealed class E extends D permits C { }
+
+    sealed class C extends E permits A { }
+
+    // violation below, 'Usage of type 'C' is not allowed.'
+    final class A extends C { }
+
+    // violation below, 'Usage of type 'D' is not allowed.'
+    final class B extends D { }
+}

